internal/datastore/postgres/gc.go

```diff
@@ -0,0 +1,145 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgtype"
+	"github.com/rs/zerolog/log"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+)
+
+func (pgd *pgDatastore) runGarbageCollector() error {
+	log.Info().Dur("interval", pgd.gcInterval).Msg("garbage collection worker started for postgres driver")
+
+	for {
+		select {
+		case <-pgd.gcCtx.Done():
+			log.Info().Msg("shutting down garbage collection worker for postgres driver")
+			return pgd.gcCtx.Err()
+
+		case <-time.After(pgd.gcInterval):
+			err := pgd.collectGarbage()
+			if err != nil {
+				log.Warn().Err(err).Msg("error when attempting to perform garbage collection")
+			} else {
+				log.Debug().Msg("garbage collection completed for postgres")
+			}
```

Comment:
might actually want to copy that backoff v4 exponential backoff pattern here for gc

Reply:
Open an issue. This PR didn't change GC at all.